frontend/src/components/FilterMenu.js

```diff
@@ -0,0 +1,325 @@
+
+  const onRemove = () => {
+    onRemoveFilter(id);
+  };
+
+  let readableFilterName = '';
+  if (selectedTopic) {
+    readableFilterName = selectedTopic.display;
+  }
+
+  const buttonAriaLabel = readableFilterName
+    ? `remove ${readableFilterName} ${condition} ${query} filter. click apply filters to make your changes`
+    : 'remove this filter. click apply filters to make your changes';
+
+  return (
+    <li className="ttahub-filter-menu-item display-flex">
+      <select
+        name="topic"
+        aria-label="topic"
+        value={topic}
+        onChange={(e) => onUpdate(e.target.name, e.target.value)}
+        className="usa-select margin-right-1"
+        ref={firstSelect}
+      >
+        <option value="">- Select -</option>
+        {possibleFilters.map(({ id: filterId, display }) => (
+          <option key={filterId} value={filterId}>{display}</option>
+        ))}
+      </select>
+      <select
+        name="condition"
+        aria-label="condition"
+        value={condition}
+        onChange={(e) => onUpdate(e.target.name, e.target.value)}
+        className="usa-select"
+      >
+        <option value="">- Select -</option>
+        {conditions.map((c) => <option key={c} value={c}>{c}</option>)}
+      </select>
+      { selectedTopic && condition
+        ? selectedTopic.renderInput()
+        : <DummySelect /> }
+      <button
+        type="button"
+        aria-label={buttonAriaLabel}
+        className="usa-button usa-button--unstyled font-sans-xs margin-right-1 margin-left-0"
+        onClick={onRemove}
+      >
+        <FontAwesomeIcon color="gray" icon={faTimesCircle} />
+      </button>
+    </li>
+  );
+}
+
+FilterItem.propTypes = {
+  filter: filterProp.isRequired,
+  onRemoveFilter: PropTypes.func.isRequired,
+  onUpdateFilter: PropTypes.func.isRequired,
+};
+
+/**
+ * Renders the interior of the menu, the set of FilterItems
+ * @param {Object} props
+ * @returns JSX Object
+ */
+export function Menu({
+  filters, onApplyFilters, toggleMenu, hidden,
+}) {
+  const [items, setItems] = useState([...filters]);
+
+  const onApply = () => {
+    onApplyFilters(items);
+  };
+
+  const onRemoveFilter = (id) => {
+    const newItems = [...items];
+    const index = newItems.findIndex((item) => item.id === id);
+
+    if (index !== -1) {
+      newItems.splice(index, 1);
+      setItems(newItems);
+    }
+  };
+
+  const onUpdateFilter = (id, name, value) => {
+    const newItems = [...items];
+    const toUpdate = newItems.find((item) => item.id === id);
+    toUpdate[name] = value;
+
+    if (name === 'topic') {
+      toUpdate.query = '';
+    }
+
+    setItems(newItems);
+  };
+  const onAddFilter = () => {
+    const newItems = [...items];
+    const newItem = {
+      id: uuidv4(),
+      display: '',
+      conditions: [],
+      renderInput: () => <span />,
+    };
+    newItems.push(newItem);
+    setItems(newItems);
+  };
+
+  const closeOnEsc = (e) => {
```

Comment:
Do also want to close the menu on blur? Also something for a different PR, but it seems like we have a lot of these menus that should function the same way. May be time to extract them all out into a single component so we don't have to worry about adding all features to every instance.